src/components/Checkbox/Checkbox.tsx

```diff
@@ -21,64 +15,59 @@ export class Checkbox extends React.Component<ICheckboxProps, ICheckboxState> {
   constructor(props: ICheckboxProps) {
     super(props);
 
-    this.state = {
-      isChecked: props.isChecked
-    };
-
-    this._id = `checkbox-${ _instance++ }`;
-
-    this._handleInputChange = this._handleInputChange.bind(this);
-  }
-
-  public componentWillReceiveProps(newProps: ICheckboxProps) {
-    if (newProps.isChecked !== this.props.isChecked) {
-      this.setState({
-        isChecked: newProps.isChecked
-      });
-    }
+    this._id = getId('checkbox-');
+    this._onChange = this._onChange.bind(this);
   }
 
   public render() {
-    let { text, isEnabled, className } = this.props;
-    let { isChecked } = this.state;
+    let {
+      checked,
+      className,
+      defaultChecked,
+      disabled,
+      label,
+      inputProps
+    } = this.props;
 
     return (
       <div className={ css('ms-ChoiceField', className) }>
         <input
-          ref={ (c): HTMLInputElement => this._checkBox = c }
+          { ...inputProps }
```

Comment:
Does React throw a warning about duplicate props if disabled is defined in inputProps and below?